### PR TITLE
fix: improve upload component scrolling issue during form validation 

### LINKED ISSUE #28869: scrollToFirstError for Upload input
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/scroll-to-the-first-error-with-upload-antd-00edf)

### Steps to reproduce
when i test with the upload item it doesn't scroll 
but when i hide it the scroll work 
event i'have added the id

### What is expected?
to scroll to the first error ( the upload input)

### What is actually happening?
it doesn't scroll

| Environment | Info |
|---|---|
| antd | 4.10.2 |
| React | 16.8.6 |
| System | Mac OS |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix: #28869

### 💡 Background and solution

一个历史遗漏问题， 表单滚动到错误位置需要在表单控件中绑定id， 但是上传组件的 id 在一个隐藏的 input 上，所以 api 无法得到元素渲染的具体位置。所以需要将收入控件渲染在 dom 中，然后用 css 内联样式隐藏起来。 

如果表单中有上传组件，则推荐 `scrollToFirstError` 以对象方式设置元素块在视口中居中显示。

```jsx
scrollToFirstError={{
  block: 'center',
}}
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    improve upload component scrolling issue during form validation        |
| 🇨🇳 Chinese |    修复上传组件在表单滑动至校验失败位置错误       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at fcc79c3</samp>

This pull request updates the `rc-upload` dependency and modifies the `Upload` component to fix a Safari bug and improve the style consistency and flexibility. It changes the files `components/upload/style/index.ts`, `components/upload/Upload.tsx`, and `package.json`.

### <samp>🤖 Generated by Copilot at fcc79c3</samp>

*  Fix upload button not clickable on Safari by adding custom styles and classNames to `rc-upload` component ([link](https://github.com/ant-design/ant-design/pull/45410/files?diff=unified&w=0#diff-a2e7152a9f97730a600f9bda0bf15029cce0148818a6ee8d2a037fafc101427dR353-R358), [link](https://github.com/ant-design/ant-design/pull/45410/files?diff=unified&w=0#diff-647a2b5c69bacca02d0a7fc39f44e3749ec7e7583a83df0259315542f6f49668L31-R44))
* Update `rc-upload` dependency version to 4.4.0 in `package.json` to use the latest bug fixes and improvements ([link](https://github.com/ant-design/ant-design/pull/45410/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L157-R157))

### CODE REVIEW

File: components/upload/Upload.tsx [hunk 1]
```diff
@@ -350,6 +350,12 @@ const InternalUpload: React.ForwardRefRenderFunction<UploadRef, UploadProps> = (
     disabled: mergedDisabled,
     beforeUpload: mergedBeforeUpload,
     onChange: undefined,
+    styles: {
+      input: { display: 'unset', visibility: 'hidden' },
```
Comment (afc163) [inline components/upload/Upload.tsx:354]: 这两个样式为啥不放 style/index.ts 里面？

Comment (Wxh16144) [reply components/upload/Upload.tsx:354]: rc-upload 里面默认存在 display none 逻辑，https://github.com/react-component/upload/pull/518/files

File: components/upload/style/index.ts [hunk 2]
```diff
@@ -28,12 +28,18 @@ const genBaseStyle: GenerateStyle<UploadToken> = (token) => {
...
-          cursor: 'pointer',
-        },
+      },
+
+      [`${componentCls}-trigger`]: {
+        cursor: 'pointer',
+        position: 'absolute',
+        inset: 0,
+        width: 0,
+        height: 0,
```
Comment (Wxh16144) [inline components/upload/style/index.ts:42]: 这个 PR 需要再等等，这里把宽和高设置 0 不太合适，如果 upload 在浏览器底部，定位会有问题。

Comment (Wxh16144) [reply components/upload/style/index.ts:42]: Draft PR 先挂一会儿，别催我😭~

**Comment by Wxh16144** [comment]
https://github.com/ant-design/ant-design/blob/287dd699222c3272490288324c9671e57d9e2bf0/components/form/style/index.ts#L87-L89 这段代码应该可以删除了 @MadCcc  review 一下? blame 是你2年前的提交

**Comment by Wxh16144** [comment]
ready~

File: components/upload/Upload.tsx (hunk 1 above)
Comment (crazyair) [reply components/upload/Upload.tsx:359]: 那要不要改 rc-upload？rc-upload 再加个滚动 demo

Comment (Wxh16144) [reply components/upload/Upload.tsx:359]: rc-upload 没有form 滚动错误的操作。 验证这个 PR 估计得本地验证了。从 issue copy 了一份可以复现代码

```tsx
import React from 'react';
import {
  Form,
  Select,
  InputNumber,
  Switch,
  Radio,
  Slider,
  Button,
  Upload,
  Rate,
} from 'antd';
import { UploadOutlined, InboxOutlined } from '@ant-design/icons';

const { Option } = Select;
const formItemLayout = {
  labelCol: {
    span: 6,
  },
  wrapperCol: {
    span: 14,
  },
};

const MUpload = (props) => {
  const { id, ..._props } = props;

  return (
    <span id={id}>
      <Upload {..._props}>{props.children}</Upload>
    </span>
  );
};

const normFile = (e) => {
  console.log('Upload event:', e);

  if (Array.isArray(e)) {
    return e;
  }

  return e && e.fileList;
};

const Demo = () => {
  const onFinish = (values) => {
    console.log('Received values of form: ', values);
  };

  return (
    <Form
      {...formItemLayout}
      onFinish={onFinish}
scrollToFirstError={{
  block: 'center',
}}
      initialValues={{
        'input-number': 3,
        'checkbox-group': ['A', 'B'],
        rate: 3.5,
      }}
    >

      <Form.Item
        wrapperCol={{
          span: 12,
          offset: 6,
        }}
      >
        <Button type="primary" htmlType="submit">
          Submit start
        </Button>
      </Form.Item>

      <br />
      <Form.Item label="Plain Text">
        <span className="ant-form-text">China</span>
      </Form.Item>
      <Form.Item
        name="select"
        label="Select"
        hasFeedback
      // rules={[
      //   {
      //     required: true,
      //     message: 'Please select your country!',
      //   },
      // ]}
      >
        <Select placeholder="Please select a country">
          <Option value="china">China</Option>
          <Option value="usa">U.S.A</Option>
        </Select>
      </Form.Item>

      <Form.Item
        name="upload"
        label="Upload"
        valuePropName="fileList"
        getValueFromEvent={normFile}
        rules={[
          {
            required: true,
            message: 'Please select your country!',
          },
        ]}
        extra="longgggggggggggggggggggggggggggggggggg"
      >
        <MUpload id="upload" name="logo" action="/upload.do" listType="picture">
          <Button>
            <UploadOutlined />
            点击上传
          </Button>
        </MUpload>
      </Form.Item>

      <Form.Item
        name="select-multiple"
        label="Select[multiple]"
      // rules={[
      //   {
      //     required: true,
      //     message: 'Please select your favourite colors!',
      //     type: 'array',
      //   },
      // ]}
      >
        <Select mode="multiple" placeholder="Please select favourite colors">
          <Option value="red">Red</Option>
          <Option value="green">Green</Option>
          <Option value="blue">Blue</Option>
        </Select>
      </Form.Item>

      <Form.Item label="InputNumber">
        <Form.Item name="input-number" noStyle>
          <InputNumber min={1} max={10} />
        </Form.Item>
        <span className="ant-form-text"> machines</span>
      </Form.Item>

      <Form.Item name="switch" label="Switch" valuePropName="checked">
        <Switch />
      </Form.Item>

      <Form.Item name="slider" label="Slider">
        <Slider
          marks={{
            0: 'A',
            20: 'B',
            40: 'C',
            60: 'D',
            80: 'E',
            100: 'F',
          }}
        />
      </Form.Item>

      <Form.Item name="radio-group" label="Radio.Group">
        <Radio.Group>
          <Radio value="a">item 1</Radio>
          <Radio value="b">item 2</Radio>
          <Radio value="c">item 3</Radio>
        </Radio.Group>
      </Form.Item>

      <Form.Item
        name="radio-button"
        label="Radio.Button"
      // rules={[
      //   {
      //     required: true,
      //     message: 'Please pick an item!',
      //   },
      // ]}
      >
        <Radio.Group>
          <Radio.Button value="a">item 1</Radio.Button>
          <Radio.Button value="b">item 2</Radio.Button>
          <Radio.Button value="c">item 3</Radio.Button>
        </Radio.Group>
      </Form.Item>

      <Form.Item name="rate" label="Rate">
        <Rate />
      </Form.Item>

      <Form.Item label="Dragger">
        <Form.Item
          name="dragger"
          valuePropName="fileList"
          getValueFromEvent={normFile}
          noStyle

          rules={[
            {
              required: true,
              message: 'Please select your country!',
            },
          ]}
        >
          <Upload.Dragger name="files" action="/upload.do">
            <p className="ant-upload-drag-icon">
              <InboxOutlined />
            </p>
            <p className="ant-upload-text">
              Click or drag file to this area to upload
            </p>
            <p className="ant-upload-hint">
              Support for a single or bulk upload.
            </p>
          </Upload.Dragger>
        </Form.Item>

      </Form.Item>
      <Form.Item label="Plain Text">
          <span style={{ display: 'inline-block', height: 200 }} className="ant-form-text">China</span>
        </Form.Item>

        <Form.Item
        wrapperCol={{
          span: 12,
          offset: 6,
        }}
      >
        <Button type="primary" htmlType="submit">
          Submit end
        </Button>
      </Form.Item>
    </Form>
  );
};

export default Demo;

```

Comment (crazyair) [reply components/upload/Upload.tsx:359]: 不一定是 Form 的 scrollToFirstError，可以自己写个滚动的方法，验证

Comment (Wxh16144) [reply components/upload/Upload.tsx:359]: 滚动没啥问题，这个 pr 解决的还是 upload 的 input 是一个 display: none  的问题

**Comment by Wxh16144** [comment]
image

有点问题， 视觉回归居然没检查出来

**Comment by Wxh16144** [comment]
我估计的 Preview Deploy 挂了是因为之前的 surge key 部署的不能对其进行覆盖导致的🤔

**Comment by Wxh16144** [comment]
我看看 jsdom 能不能模拟滚动写个单元测试

**Comment by zombieJ** [comment]
感觉可以优化一下 Form 的逻辑，先看看子元素支不支持 `ref`，然后如果支持先用 `ref` 滚动，不行再 `id` 滚动

**Pull request closed without merging**